### PR TITLE
Consolidate Unicode code-point predicates in code_point module

### PR DESCRIPTION
**Pull request by sergey-shandar**
## Summary

Moves Unicode code-point classification predicates (`isBmpCodePoint`, `isHighSurrogate`, `isLowSurrogate`, `isSupplementaryPlane`, `isValidCodePoint`) from UTF-8 and UTF-16 modules into the shared `code_point` module. These predicates describe fixed Unicode ranges, not encoding-specific details, so they belong alongside the existing shared contracts (`errorMask` and `decoder`).

## Key Changes

- **Consolidate predicates in `code_point/module.f.ts`**: Added all five code-point classification predicates derived from a single set of boundary constants (`surrogateMin`, `lowSurrogateMin`, `surrogateMax`, `bmpMax`, `maxCodePoint`). This ensures the surrogate bounds (`0xD800`–`0xDFFF`) and maximum code point (`0x10FFFF`) are defined exactly once.

- **Add comprehensive proof coverage**: Created `code_point/proof.f.ts` with test cases for each predicate, covering boundary conditions and both true/false branches.

- **Update UTF-16 module**: Removed local definitions of `isBmpCodePoint`, `isHighSurrogate`, `isLowSurrogate`, and `isSupplementaryPlane`; now imports them from `code_point`.

- **Update UTF-8 module**: Removed local definitions of `bigRange`, `smallRange`, and `isValidCodePoint`; now imports `isValidCodePoint` from `code_point` and re-exports it to maintain API stability for external consumers (`fs/mime`).

- **Update documentation**: Added section to `fs/text/README.md` explaining the consolidation and the re-export decision. Updated module docstrings to reflect the expanded scope of shared contracts.

## Implementation Details

- All predicates are built from shared boundary constants, eliminating duplication of magic numbers like `0xD800`, `0xDFFF`, and `0x10FFFF`.
- The `0xffffffff` out-of-range sentinel in UTF-16 is left unchanged (it serves a different purpose than the shared predicates).
- `isValidCodePoint` is re-exported from `utf8` rather than dropped to preserve the public API for existing external consumers.

https://claude.ai/code/session_01S27igZkZS5UrbiVvE4bXGf

### CODE REVIEW

File: fs/text/utf8/module.f.ts [hunk 1]
```diff
@@ -6,10 +6,14 @@
...
-import { decoder, errorMask } from '../code_point/module.f.ts'
+import { decoder, errorMask, isValidCodePoint } from '../code_point/module.f.ts'
 import { msb, u8List, length, type Vec } from '../../types/bit_vec/module.f.ts'
 import { codePointListToString } from '../utf16/module.f.ts'
-import { contains } from '../../types/range/module.f.ts'
+
+// `isValidCodePoint` lives in `code_point` (the shared Unicode contract); it is
+// re-exported here to keep utf8's public API stable for `fs/mime`, which imports
+// it from this module.
+export { isValidCodePoint }
```
Comment (sergey-shandar) [inline fs/text/utf8/module.f.ts]: We don't need to re-export. Our strategy is to make breaking changes if they are right and add them to CHANGELOG as `**BREAKING CHANGES:**`. BTW, add this rule to `AGENTS.md`.

Comment (sergey-shandar) [reply fs/text/utf8/module.f.ts]: Done. Dropped the re-export — `isValidCodePoint` now lives solely on `code_point`, and `fs/mime` imports it from there. Added a `**BREAKING CHANGES:**` CHANGELOG entry and the breaking-change strategy rule to `AGENTS.md`. tsc + all 2002 proofs pass.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by sergey-shandar** [comment]
@codex review

**Comment by sergey-shandar** [comment]
@codex review

File: fs/text/code_point/module.f.ts [hunk 2]
```diff
@@ -36,3 +39,63 @@ export const decoder = <Unit, S, Cp>(
...
+ * (`0xD800`–`0xDFFF`) splits into a high half (`0xD800`–`0xDBFF`) and a low half
+ * (`0xDC00`–`0xDFFF`); the BMP ends at `0xFFFF` and `maxCodePoint` (`0x10FFFF`)
+ * is the largest assignable code point. Every predicate below is derived from
+ * these constants so the surrogate bounds and the maximum appear exactly once.
+ */
+const surrogateMin = 0xd800
+const lowSurrogateMin = 0xdc00
+const surrogateMax = 0xdfff
+const bmpMax = 0xffff
+const maxCodePoint = 0x10_ffff
```
[the remark this answers is not in the document]
Comment (sergey-shandar) [reply fs/text/code_point/module.f.ts]: Fixed in fcc37ae — pinned all five boundary constants with `as const` so they keep their literal types instead of widening to `number`. tsc passes.

---
_Generated by [Claude Code](https://claude.ai/code)_

[the remark this answers is not in the document]
Comment (sergey-shandar) [reply fs/text/code_point/module.f.ts]: Fixed in fcc37ae — pinned all five boundary constants with `as const` (duplicate of the thread above).

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by sergey-shandar** [comment]
@codex review

File: fs/text/code_point/todo/share-code-point-predicates.md [hunk 3]
```diff
@@ -1,106 +0,0 @@
-## Share Unicode code-point predicates in `code_point`
```
[the remark this answers is not in the document]
Comment (sergey-shandar) [reply fs/text/code_point/todo/share-code-point-predicates.md:1]: Fixed in 8d9a37d — replaced all three links to the deleted issue in `fs/mime/todo/text-vs-binary-control-bytes.md` (lines 40, 81, 96) with references to the now-existing shared home (`fs/text/code_point/module.f.ts` and `fs/text/README.md`), preserving the design context. The only remaining mention is the plain-text issue slug in the CHANGELOG, which AGENTS.md permits.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by sergey-shandar** [comment]
@codex review
